4_other_sources/src/get_norwest_data.R

```diff
@@ -0,0 +1,178 @@
+    as_promises = TRUE)
+
+  loop_tasks(task_plan, '4_norwest_datafile_tasks.yml')
+  combine_to_ind(out_file, c('4_other_sources/out/norwest_raw_temp_data.feather', '4_other_sources/out/norwest_raw_site_data.rds'))
+  #scmake('raw_norwest_data.rds.ind_promise', remake_file='4_norwest_datafile_tasks.yml')
+}
+# download and unzip file, return rds
+fetch_data_files <- function(base_url, files, region, tmp_loc = '4_other_sources/tmp') {
+  # create url for download
+  file_name <- files$data_files[files$tasks %in% region]
+  temp_url <- file.path(base_url, file_name)
+  order <- which(files$tasks %in% region)
+  # create a temporary space to download zip
+  ext_directory <- file.path('4_other_sources/tmp', region)
+  
+  if (!file.exists(ext_directory)) {
+    message(sprintf("    %d of %d: Downloading Norwest temperature file %s", order, nrow(files), file_name))
+    download.file(temp_url, ext_directory)
+  } 
+ 
+  # unzip to tmp file location
+  unzip(ext_directory, exdir = '4_other_sources/tmp')
+  
+
+  # read in data
+  outfile <- gsub('.zip', '', gsub('.*/', '', file_name))
+  outfile <- list.files(tmp_loc)[grepl(outfile, list.files(tmp_loc))]
+
+  if (grepl('xlsx', outfile)) {
+    dat <- read_xlsx(file.path(tmp_loc, outfile))
+  } else if (grepl('txt', outfile)) {
+    dat <- readr::read_delim(file.path(tmp_loc, outfile), delim = ',') %>%
+      mutate(SampleDate = as.Date(SampleDate, format = '%m/%d/%Y'))
+      
+  }
+  dat <- mutate(dat, region = region)
+  return(dat)
+}
+
+fetch_site_files <- function(base_url, files, region, tmp_loc = '4_other_sources/tmp') {
+  # create url for download
+  file_name <- files$site_files[files$tasks %in% region]
+  temp_url <- file.path(base_url, file_name)
+  order <- which(files$tasks %in% region)
+  # create a temporary space to download zip
+  ext_directory <- file.path('4_other_sources/tmp', paste0(region, '_sites'))
+  
+  if (!file.exists(ext_directory)) {
+    message(sprintf("    %d of %d: Downloading Norwest site file %s", order, nrow(files), file_name))
+    download.file(temp_url, ext_directory)
+  } 
+  # unzip to tmp file location
+  unzip(ext_directory, exdir = '4_other_sources/tmp')
+  
+  # read in data
+  outfile <- gsub('.zip', '', gsub('.*/', '', file_name))
+  
+  # annoying name change for region NorthernCaliforniaCoastalKlamath
+  if (region == "NorthernCaliforniaCoastalKlamath") {
+    outfile <- 'NorWeST_ObservedTempPoints_NorthCalifCoastalKlamath'
+  }
+  # read in shape file
+  geo_dat <- sf::st_read(file.path('4_other_sources/tmp', paste0(outfile, '.shp'))) %>%
+    mutate(region = region)
+  
+  return(geo_dat)
+}
+
+namedList <- function(...) {
+  L <- list(...)
+  snm <- sapply(substitute(list(...)),deparse)[-1]
+  if (is.null(nm <- names(L))) nm <- snm
+  if (any(nonames <- nm=="")) nm[nonames] <- snm[nonames]
+  setNames(L,nm)
+}
+combine_temp_data <- function(ind_file, ...) {
+
+  data_list <- namedList(...)
```

Comment:
This seems complicated. I think you can use `list()` in place of `namedList()`. Here's a test/demo:
```r
namedList <- function(...) {
  L <- list(...)
  snm <- sapply(substitute(list(...)),deparse)[-1]
  if (is.null(nm <- names(L))) nm <- snm
  if (any(nonames <- nm=="")) nm[nonames] <- snm[nonames]
  setNames(L,nm)
}
f <- function(...) {
  list(...)
}
g <- function(...) {
  namedList(...)
}
```
```
> f(x=8, y=2)
$x
[1] 8

$y
[1] 2

> g(x=8, y=2)
$x
[1] 8

$y
[1] 2
```

Reply:
The problem is the ... are object names, so in your example above, I would be inputting the arguments like this, and I need the names of the objects to sort downstream:
![image](https://user-images.githubusercontent.com/15788176/94966273-9c74d100-04c2-11eb-9362-741c51861e9b.png)

Reply:
Ohhh, yep.